### ISSUE TITLE
Move Help into nash-help module

### ISSUE THREAD

**Issue by scottcorgan** [scottcorgan/nash#18]
Including this module into Nash will add all help methods to prototypes:
- `usage()`
- `description()`
- `hidden()`
- etc.

**Comment by vohof** [comment]
`nash-help` does not exist.

**Comment by scottcorgan** [comment]
You're right. Working on getting that implanted in the next few days. I'll let you know as soon as it's up.

**Comment by indexzero** [comment]
@scottcorgan any progress on this? I saw `nash-usage` on your Github profile but it is largely an empty repository. I'd really like to rewrite the `forever` CLI with `nash`, but CLI-based help is a huge part of it.

`nash` stands out compared to `commander` because of the per command flag support. Which is going to be incredibly helpful for `forever` users if it can also generate the per command help since this is where the majority of confusion arises.

So ... how can I help?

**Comment by scottcorgan** [comment]
There is much progress on this. I'll detail in the morning. I'm interested
in how you plan to use it. It's always good to have some objective views.

**Comment by indexzero** [comment]
If you're interested in the nuts and bolts you can check out the forever CLI code: https://github.com/foreverjs/forever/blob/master/lib/forever/cli.js

**Comment by scottcorgan** [comment]
Ah, yes, Flatiron. Inspiration for Nash for sure. The details for this plugin has more to do with the new api for Nash, and it's moving to an org with a rename.

The new api is focused on composability, so something like a usage screen is easy to plugin at any point. I've built a pretty large cli for Divshot (https://github.com/divshot/divshot-cli), and that's where Nash came from. After awhile, we started seeing uses for individual commands outside of the full CLI. Composability would have been amazing (for more than just packages, but cli packages).

With that said, renaming Nash to `cmd` and putting it under and org seemed like the best next move for Nash. If you want to take a look at  https://github.com/cmd-js/cmd/issues/2 for discussion on that new api. We also have been chatting on Gitter: https://gitter.im/cmd-js/cmd.

I have the core done and I am working on this first plugin now. (Still pre-alpha)

I want to get this right for apps like forever and divshot-cli. So I'm interested in real world, objective input.
